src/config/field-rules.js

```diff
@@ -66,6 +67,60 @@ export default new Map([
                 elseValue: false,
             }],
         },
+        {
+            field: 'optionSet',
+            when: {
+                operator: 'PREDICATE',
+                value: () => {
+                    // value can be ID or model here, so need to check for valueType in operation...
+                    return true
+                }
+            },
+            operations: [{
+                type: 'CHANGE_VALUE',
+                field: 'valueType',
+                setValue: (model, fieldConfig) => {
+                    const isMultiText = model && (model.valueType === 'MULTI_TEXT'
+                            || (model.optionSet && model.optionSet.valueType === 'MULTI_TEXT'));
+
+                    if(isMultiText) {
+                        // this is needed for the UI to reflect that valueType is MULTI_TEXT
+                        // However it is always disabled in this case, because it's dictated by the optionSet
+                        // see setValueTypeToOptionSet()
+                        if(fieldConfig.originalOptions) {
+                            fieldConfig.props.options = fieldConfig.originalOptions;
+                        }
+                        if(model.domainType === 'TRACKER') {
+                            // only set fieldConfig here, model is set below in optionSet since it depends on
+                            // the optionSet value to be set to check for the valueType.
+                            fieldConfig.value = undefined
+                        }
+                } else {
+                        // MULTI_TEXT should not be selectable for valueType for dataElements, since it needs to be an optionSet
+                        // thus the option is removed from the valueType options
+                        const options = fieldConfig.props.options;
+                        if(!fieldConfig.originalOptions) {
+                            fieldConfig.originalOptions = fieldConfig.props.options;
+                        }
+                        fieldConfig.props.options = options.filter(option => option.value !== 'MULTI_TEXT');
+                    }
+                },
+            }, {
+                type: 'CHANGE_VALUE',
+                field: 'optionSet',
+                setValue: (model, fieldConfig) => {
+                    const isMultiText = model && (model.valueType === 'MULTI_TEXT'
```

Comment:
would it just be sufficient to check if the optionSet is `MULTI_TEXT` here? It seems like you cannot ever directly select `MULTI_TEXT`, so it's only being set indirectly when you choose a `MULTI_TEXT` optionSet?

Reply:
You would think so but it's needed when editing a data Element, since the `optionSet`-object isn't fully loaded at first load - it just knows the id - optionSets are loaded afterwards.